pci/src/vfio.rs

```diff
@@ -275,6 +275,41 @@ pub struct MmioRegion {
     pub(crate) index: u32,
     pub(crate) user_memory_regions: Vec<UserMemoryRegion>,
 }
+
+trait MmioRegionRange {
+    fn check_range(&self, guest_addr: u64, size: u64) -> bool;
+    fn find_user_address(&self, guest_addr: u64) -> Option<u64>;
+}
+
+impl MmioRegionRange for Vec<MmioRegion> {
+    // Check if a guest address is within the range of mmio regions
+    fn check_range(&self, guest_addr: u64, size: u64) -> bool {
+        for region in self.iter() {
+            if guest_addr >= region.start.raw_value()
+                && guest_addr.checked_add(size).unwrap()
+                    <= region.start.raw_value().checked_add(region.length).unwrap()
```

Comment:
Similarly, it would be good to avoid using unwrap(). You can return `false` for when `checked_add()` failed.

Reply:
Done, this should now return `false` if `checked_add()` fails